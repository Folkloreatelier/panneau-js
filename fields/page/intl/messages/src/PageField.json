[
  {
    "id": "fields.page.placeholder",
    "description": "Placeholder of the page autosuggest input",
    "defaultMessage": "Search for a page..."
  }
]