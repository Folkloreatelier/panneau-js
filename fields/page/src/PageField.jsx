import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { defineMessages } from 'react-intl';
import { PropTypes as PanneauPropTypes, withUrlGenerator } from '@panneau/core';
import ItemField from '@panneau/field-item';

const messages = defineMessages({
    placeholder: {
        id: 'fields.page.placeholder',
        description: 'Placeholder of the page autosuggest input',
        defaultMessage: 'Search for a page...',
    },
});

const propTypes = {
    urlGenerator: PanneauPropTypes.urlGenerator,
    placeholder: PanneauPropTypes.message,
    resourceId: PropTypes.string,
    endpoint: PropTypes.string,
};

const defaultProps = {
    urlGenerator: null,
    placeholder: messages.placeholder,
    resourceId: 'pages',
    endpoint: null,
};

const contextTypes = {
    definition: PanneauPropTypes.definition,
};

class PageField extends PureComponent {
    getEndpoint() {
        const { resourceId, urlGenerator, endpoint } = this.props;
        const { definition } = this.context;
        if (endpoint !== null) {
            return endpoint;
        }
        if (urlGenerator === null || definition === null) {
            return null;
        }
        const resources = definition.resources || [];
        const resource = resources.find(it => it.id === resourceId) || null;
        if (resource === null) {
            return null;
        }
        return urlGenerator.route(
            typeof resource.routes !== 'undefined'
                ? `resources.${resources.id}.index`
                : 'resources.index',
            {
                resource: resource.id,
            },
        );
    }

    render() {
        return (
            <ItemField
                cardItemMap={{
                    name: 'title',
                    created_at: 'created_at',
                    thumbnail: 'image',
                }}
                autosuggestProps={{
                    suggestionsEndpoint: this.getEndpoint(),
                    suggestionValuePath: 'title',
                    suggestionTitlePath: 'title',
                }}
                {...this.props}
            />
        );
    }
}

PageField.propTypes = propTypes;
PageField.defaultProps = defaultProps;
PageField.contextTypes = contextTypes;

export default withUrlGenerator()(PageField);
