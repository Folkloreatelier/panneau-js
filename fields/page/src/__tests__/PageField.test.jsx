/* eslint-disable import/no-extraneous-dependencies */
import React from 'react';
import renderer from 'react-test-renderer';
// import { shallow, mount } from 'enzyme';
// import sinon from 'sinon';
import { IntlProvider } from 'react-intl';
import PageField from '../PageField';

test('match snapshot', () => {
    const component = renderer.create((
        <IntlProvider locale="en">
            <PageField
                label="Label"
                value={null}
            />
        </IntlProvider>
    ));
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

// test('value is in <input />', () => {
//     const field = shallow(<PageField value="test" />);
//     expect(field.find('input').prop('value')).toEqual('test');
//
//     const nullField = shallow(<PageField value={null} />);
//     expect(nullField.find('input').prop('value')).toEqual('');
// });
