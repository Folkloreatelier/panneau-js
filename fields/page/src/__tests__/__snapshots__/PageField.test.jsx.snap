// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`match snapshot 1`] = `
<div
  className="form-group undefined"
>
  <div
    className="form-group-label"
  >
    <label
      className="control-label"
      htmlFor={null}
    >
      Label
    </label>
  </div>
  <div
    className="form-group-inner"
    style={
      Object {
        "display": "block",
      }
    }
  >
    <div>
      <div
        className="form-group undefined"
      >
        <div
          className="form-group-inner"
          style={
            Object {
              "display": "block",
            }
          }
        >
          <div
            aria-expanded={false}
            aria-haspopup="listbox"
            aria-owns="react-autowhatever-1"
            className="undefined"
            role="combobox"
          >
            <input
              aria-activedescendant={null}
              aria-autocomplete="list"
              aria-controls="react-autowhatever-1"
              autoComplete="off"
              className="form-control undefined"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              placeholder="Search for a page..."
              type="text"
              value=""
            />
            <div
              className="undefined"
              id="react-autowhatever-1"
              role="listbox"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
